app/helpers/application_helper/button/basic.rb

```diff
@@ -64,6 +65,9 @@ def skipped?
 
   # Tells whether the button should displayed in the toolbar or not
   def visible?
+    # hide select power management buttons for Openstack::InfraManager
+    return false if %w(host_standby host_shutdown host_reboot host_reset).include?(@child_id) &&
+      @record.kind_of?(ManageIQ::Providers::Openstack::InfraManager)
```

Comment:
Ah, this can't be here, it's a 'pluggable providers concern'

You should add supports for each feature, to the model having the feature (OS Infra Host?) Then test for the supports, without mentioning ManageIQ::Providers::Openstack::InfraManager

You will probably need the specific button class here. @durandom @martinpovolny we can't probably test the supports in the base class, right? Maybe creating SupportsFeatureButton class for now?